src/js/view/utils/resize-listener.js

```diff
@@ -1,44 +1,61 @@
 import { requestAnimationFrame, cancelAnimationFrame } from 'utils/request-animation-frame';
 import { createElement } from 'utils/dom';
-import { css, style } from 'utils/css';
+import { style } from 'utils/css';
 
-const topLeft = {
-    display: 'block',
-    position: 'absolute',
-    top: 0,
-    left: 0
-};
+const instances = [];
+let resizeRaf = -1;
 
-const stretch = {
-    width: '100%',
-    height: '100%'
-};
-
-let contractTriggerCssAdded = false;
-
-const lazyAddCss = function() {
-    if (!contractTriggerCssAdded) {
-        contractTriggerCssAdded = true;
-        css('.jw-contract-trigger::before', Object.assign({
-            content: '',
-            overflow: 'hidden',
-            width: '200%',
-            height: '200%'
-        }, topLeft));
-    }
-};
+function scrollListener() {
+    cancelAnimationFrame(resizeRaf);
+    resizeRaf = requestAnimationFrame(() => {
+        instances.forEach(resizeListener => {
+            resizeListener.view.updateBounds();
+            const width = resizeListener.view.model.get('containerWidth');
+            resizeListener.resized = resizeListener.width !== width;
+            resizeListener.width = width;
+        });
+        instances.forEach(resizeListener => {
+            resizeListener.contractElement.scrollLeft = resizeListener.width * 2;
+        });
+        instances.forEach(resizeListener => {
+            style(resizeListener.expandChild, { width: resizeListener.width + 1 });
+            if (resizeListener.resized && resizeListener.view.model.get('visibility')) {
```

Comment:
why the viewability check if we're not updating on viewable? the scroll listener is only listening for the scroll on the internal container, couldn't we hit a case where we resize with player offscreen and then without resizing again scroll up to make the player viewable and the sizing is off?

Reply:
In view.js `_model.on('change:visibility', redraw);` calls `view.updateStyles()`. There's no need to invalidate the layout for players not in view.